### PR TITLE
MLB-1796 COLLECTION PLAN COMPONENT

### PR DESCRIPTION
**Pull request by jlplks**
# 📲 What

Create collection plan component for PLOT

# 🛠 How

Created component with compose library

# 👀 See

Screenshot 2024-11-11 at 11 49 25 a m

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+@Composable
+fun CollectionPlan(isEligible: Boolean, initialSelectedOption: String = "Pledge in full") {
+    var selectedOption by remember { mutableStateOf(initialSelectedOption) }
+
+    Column(modifier = Modifier.fillMaxWidth()) {
+        PledgeOption(
+            optionText = stringResource(id = R.string.fpo_pledge_in_full),
+            selected = selectedOption == "Pledge in full",
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt:77]: @jlplks question here what will happen with this `selectedOption == "Pledge in full"` if the user has as selected language Spanish or any other language?
We could probably define some kind of Enum class to differentiate when the selected option is `Pledge in full` vs `Pledge over time`.

File: app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+@Preview(showBackground = false, name = "Eligible - Pledge Over Time Selected")
+@Composable
+fun PreviewPledgeOverTimeSelected() {
+    KSTheme {
+        CollectionPlan(isEligible = true, initialSelectedOption = "Pledge Over Time")
+    }
+}
+
+@Preview(showBackground = false, name = "Not Eligible")
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt:52]: @jlplks nitpick here can we add the same previews as well for the darktheme?

Comment (jlplks) [reply app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt:52]: Screenshot 2024-11-12 at 3 39 48 p m

File: app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt (hunk 1 above)
Comment (jlplks) [reply app/src/main/java/com/kickstarter/ui/views/compose/checkout/CollectionPlan.kt]: Enum added to manage the Collection plan options

**Review by Arkariang — APPROVED** [review]

**Pull request merged**